### PR TITLE
Allow tamplate-haskell-2.18 in optics-th

### LINKED ISSUE #436: GHC9.2.1
It looks like GHC9.2.1 now ships `template-haskell-2.18.0.0`.

### PR DESCRIPTION
**Pull request by phadej**
Add GHC-9.2 job too, except for optics-sop, metametapost (blocked on `generics-sop`) and `template-haskell-optics`

Resolves #436

### CODE REVIEW

**Comment by phadej** [comment]
@arybczak could you look at this. There are few inspection tests failing. Feel free to push into this branch.

I'll make revision anyway, as the Core difference doesn't seem fatal.

**Comment by phadej** [comment]
ping @arybczak

**Comment by arybczak** [comment]
I'll try to have a look tomorrow ☃️

File: optics/tests/Optics/Tests/Core.hs [hunk 1]
```diff
@@ -92,7 +92,7 @@ coreTests = testGroup "Core"
   , testCase "optimized rhs13" $
     assertSuccess $(inspectTest $ hasNoProfunctors 'rhs13)
   , testCase "traverseOf_ itraversed = traverseOf_ folded" $
-    assertSuccess $(inspectTest $ 'lhs14 ==- 'rhs14)
+    ghc92failure $(inspectTest $ 'lhs14 ==- 'rhs14)
```
Comment (phadej) [inline optics/tests/Optics/Tests/Core.hs:95]: Should there be a comment why this fails?

Other seems to be known issues which is reassuring

Comment (arybczak) [reply optics/tests/Optics/Tests/Core.hs:96]: I added a comment, it's a similar issue to others.

**Pull request merged**